packages/cli/src/util/createServer.ts

```diff
@@ -48,7 +50,39 @@ export async function createSingleProcessPrism(options: CreatePrismOptions) {
   }
 }
 
-async function createPrismServerWithLogger(options: CreatePrismOptions, logInstance: Logger) {
+async function createPrismServerWithLogger(options: CreatePrismOptions & { spec: string }, logInstance: Logger) {
+  const spec = options.spec;
+  const watcher = chokidar.watch(spec);
+
+  let server = await createFastifyServerWithLogger(options, logInstance);
+
+  watcher.on('change', () => {
+    logInstance.start('Restarting Prism…');
+
+    server.fastify
+      .close()
+      .then(() => getHttpOperationsFromResource(spec))
```

Comment:
I think we should try to load the operations first, so that in case we cannot correctly parse them we can just stop processing this and be like "Yeah something is wrong so we're going to keep the server running with the original file.